deparse.c

```diff
@@ -2621,8 +2621,16 @@ sqlite_deparse_const(Const *node, deparse_expr_cxt *context, int showtype)
 			break;
 		case BITOID:
 		case VARBITOID:
-			extval = OidOutputFunctionCall(typoutput, node->constvalue);
-			appendStringInfo(buf, "B\'%s\'", extval);
+			{
+				extval = OidOutputFunctionCall(typoutput, node->constvalue);
+				if (strlen(extval) > SQLITE_FDW_BIT_DATATYPE_BUF_SIZE - 1 )
+				{
+					ereport(ERROR, (errcode(ERRCODE_FDW_INVALID_DATA_TYPE),
+							errmsg("SQLite FDW dosens't support very long bit/varbit data"),
```

Comment:
Hello @nxhai98 ! Description added in `README.md`, see https://github.com/pgspider/sqlite_fdw/pull/83/commits/77ba282682e6c158be8fd04a6b956e234554b7be
Why base affinity is `int`, not `blob`?

1. there was bad `int` implementation in our FDW
2. `blob` in SQLite is very poor affinity for available operations list. Both PostgreSQL `bit`/`varbit` and SQLite `int` ([example](https://www.tutorialspoint.com/sqlite/sqlite_bitwise_operators.htm)) take part in operations like PostgreSQL [`|` `>>` `<<` `&` `~` `#`](https://www.postgresql.org/docs/16/functions-bitstring.html) It's important because typical `bit`/`varbit` usage is some hardware register value storage.

My implementation is not very effective and based on PostgreSQL text presentation because `bit`/`varbit` is rare data type and there is length limitation - 64 bits. With this limitation there is no critical productivity difference.

Reply:
Thank for your explanation and update.    
Please help me to resolve this thread.